src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -280,11 +281,7 @@ private function parseDefinition(\DOMElement $service, $file, array $defaults =
             if ($function = $configurator->getAttribute('function')) {
                 $definition->setConfigurator($function);
             } else {
-                $configuratorService = $this->getChildren($configurator, 'service');
-
-                if (isset($configuratorService[0])) {
-                    $class = $this->parseDefinition($configuratorService[0], $file);
```

Comment:
I fixed another bug at the same time, the fact that the factories/configurators were inline meant for prototypes/instanceof conditionals that they were not always the same instance which wasn't consistent with the rest of the anonymous services.
Note that it wasn't an issue before as a service was never cloned.